test/infrastructure/docker/api/v1alpha4/dockermachinetemplate_webhook.go

```diff
@@ -41,12 +44,16 @@ func (m *DockerMachineTemplate) ValidateCreate() error {
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
-func (m *DockerMachineTemplate) ValidateUpdate(old runtime.Object) error {
-	oldCRS := old.(*DockerMachineTemplate)
-	if !reflect.DeepEqual(m.Spec, oldCRS.Spec) {
-		return errors.New("DockerMachineTemplateSpec is immutable")
+func (m *DockerMachineTemplate) ValidateUpdate(oldRaw runtime.Object) error {
+	var allErrs field.ErrorList
+	old := oldRaw.(*DockerMachineTemplate)
```

Comment:
Sorry last one. We usually seem to check if the type case works and then produce a nice looking panic (instead of the default one :P)
```bash
	if !ok {
		return apierrors.NewBadRequest(fmt.Sprintf("expected a Machine but got a %T", old))
	}
```

Reply:
(same for DockerClusterTemplate and the KubeadmControlPlaneTemplate PR)